### PR TITLE
Crystal language support

### PR DESCRIPTION
**Pull request by Jomy10**
I added syntax highlighting for the [Crystal language](https://crystal-lang.org/).

Note that some syntax highlighting is missing due to the [tree sitter](https://github.com/will/tree-sitter-crystal#status) not being complete yet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+**/.DS_Store # macOS
```
Comment (the-mikedavis) [inline .gitignore:1]: This looks unrelated: can you remove this change?

File: runtime/queries/crystal/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+; method definition/call
+(identifier) @function.method
+
+; types
+(generic_type) @type
+(union_type) @type
+(type_identifier) @type
+
+
+(local_variable) @variable.local
```
Comment (the-mikedavis) [inline runtime/queries/crystal/highlights.scm:74]:
```suggestion
(local_variable) @variable
```

File: runtime/queries/crystal/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+[
+  (nil)
+  (bool)
+] @constant.builtin
+
+[
+  (integer)
+  (float)
+] @number
```
Comment (the-mikedavis) [inline runtime/queries/crystal/highlights.scm:36]:
```suggestion
(integer) @constant.numeric.integer
(float) @constant.numeric.float
```

File: runtime/queries/crystal/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+(operator) @operator
+
+(comment) @comment
+
+; literals
+
+[
+  (nil)
+  (bool)
+] @constant.builtin
```
Comment (the-mikedavis) [inline runtime/queries/crystal/highlights.scm:31]:
```suggestion
(nil) @constant.builtin
(bool) @constant.builtin.boolean
```

File: runtime/queries/crystal/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+; variables
+
+[
+  (local_variable)
+] @variable
+
+[
+  (instance_variable)
+  (class_variable)
+] @property
```
Comment (the-mikedavis) [inline runtime/queries/crystal/highlights.scm:57]:
```suggestion
] @variable.other.member
```

**Comment by Jomy10** [comment]
I have implemented the requested changes.

File: runtime/queries/crystal/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+(symbol) @string.special.symbol
+
+(regex) @string.special.regex
+
+; variables
+
+[
+  (local_variable)
+] @variable
```
Comment (the-mikedavis) [inline runtime/queries/crystal/highlights.scm:48]: Looks like this is duplicated on the last line

**Review by the-mikedavis — APPROVED** [review]

**Comment by the-mikedavis** [comment]
Thanks!
